src/plugins/expressions/common/service/expressions_fork.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  ExpressionExecutionParams,
+  ExpressionsService,
+  ExpressionsServiceSetup,
+  ExpressionsServiceStart,
+} from '.';
+import { ExpressionAstExpression } from '../ast';
+
+export interface ExpressionServiceFork {
+  setup: () => ExpressionsServiceSetup;
+  start: () => ExpressionsServiceStart;
+}
+
+/**
+ * `ExpressionsService` class is used for multiple purposes:
+ *
+ * 1. It implements the same Expressions service that can be used on both:
+ *    (1) server-side and (2) browser-side.
+ * 2. It implements the same Expressions service that users can fork/clone,
+ *    thus have their own instance of the Expressions plugin.
+ * 3. `ExpressionsService` defines the public contracts of *setup* and *start*
+ *    Kibana Platform life-cycles for ease-of-use on server-side and browser-side.
+ * 4. `ExpressionsService` creates a bound version of all exported contract functions.
+ * 5. Functions are bound the way there are:
+ *
+ *    ```ts
+ *    registerFunction = (...args: Parameters<Executor['registerFunction']>
+ *      ): ReturnType<Executor['registerFunction']> => this.executor.registerFunction(...args);
+ *    ```
+ *
+ *    so that JSDoc appears in developers IDE when they use those `plugins.expressions.registerFunction(`.
+ */
+export class ExpressionsServiceFork implements ExpressionServiceFork {
+  /**
+   * @note Workaround since the expressions service is frozen.
+   */
+  private namespace: string | undefined;
+  private expressionsService: ExpressionsService;
+
+  constructor(namespace: string, expressions: ExpressionsService) {
+    this.namespace = namespace;
+    this.expressionsService = expressions;
+  }
+
+  /**
+   * Returns Kibana Platform *setup* life-cycle contract. Useful to return the
+   * same contract on server-side and browser-side.
+   */
+  public setup(): ExpressionsServiceSetup {
```

Comment:
- I don't think this will be extended internally and it **shouldn't** be extended externally. 
- we don't want to make the methods visible on the ExpressionServiceFork class, if we make them protected how can we test this from the outside then ?
- for me personally in this specific case readability is better as it is now, makes it obvious that the fork is just a wrapper around ExpressionsService and we don't need to do the binding

Reply:
> I don't think this will be extended internally and it shouldn't be extended externally.
Well, we cannot predict that. That's not about the extension necessity but more about abstraction. Those methods seem to be like internal API and not an implementation detail. Even though we export only `start` and `setup`, they define the forked service.

> we don't want to make the methods visible on the ExpressionServiceFork class, if we make them protected how can we test this from the outside then ?
That's possible to call them through an extended anonymous class. But in cases like that, we can just put related tests under the `describe('method')` block since it is not a private method.

> for me personally in this specific case readability is better as it is now, makes it obvious that the fork is just a wrapper around ExpressionsService and we don't need to do the binding
I think it's better if we can extract those into methods. It should not make it less straightforward, but it will be more consistent with the `ExpressionsService`. The latter makes sense because we are actually forking that service.
If we do that, it will make the decorator pattern more visible so that it still should be clear that the class is a wrapper.